### DISCUSSION TITLE
How to revoke token ?

### DISCUSSION THREAD

**Discussion by maswebber** [leafsphp/leaf#213] [Q&A]
Hello,
Anyone can help how to revoke user token in Leaf PHP API ?
in Laravel we can use something like this `$user->tokens()->delete()` or in Codeigniter 4 like this `$user->revokeAccessToken($rawToken);`

**Comment by franklinvillalobos** [comment]
It is not possible to revoke a token, leaf generates a token signed by the server.
If you want to be able to revoke the tokens as is done in Laravel, you have to create a table where you can store them and then modify the default auth service and validate the token on your own and invalidate it when you want to cancel it.
